src/program.ts

```diff
@@ -113,6 +113,49 @@ class TypeAlias {
   type: CommonTypeNode;
 }
 
+/** Represents the kind of an operator overload. */
+export enum OperatorKind {
+  INVALID,
+  INDEXED_GET,
+  INDEXED_SET,
+  ADD,
+  SUB,
+  MUL,
+  DIV,
+  REM,
+  AND,
+  OR,
+  XOR,
+  EQ,
+  NE,
+  GT,
+  GE,
+  LT,
+  LE
+}
+
+function operatorKindFromString(str: string): OperatorKind {
+  switch (str) {
+    case "[]" : return OperatorKind.INDEXED_GET;
+    case "[]=": return OperatorKind.INDEXED_SET;
+    case "+"  : return OperatorKind.ADD;
+    case "-"  : return OperatorKind.SUB;
+    case "*"  : return OperatorKind.MUL;
```

Comment:
I think we can easely add power operator '**' as well. WDYT?

Reply:
Sure, should be possible to override everything except "===" and "!==" :)